data/dataexchange/src/test/java/tech/pegasys/teku/data/SlashingProtectionRepairerTest.java

```diff
@@ -0,0 +1,220 @@
+
+  final List<String> keys =
+      List.of(
+          randomValidatorFile(),
+          randomValidatorFile(),
+          randomValidatorFile(),
+          randomValidatorFile());
+  final Map<String, Optional<ValidatorSigningRecord>> testData =
+      Map.of(
+          keys.get(0), Optional.empty(),
+          keys.get(1), optionalSigningRecord(ONE, ONE, ONE),
+          keys.get(2),
+              optionalSigningRecord(
+                  UInt64.valueOf(1024000), UInt64.valueOf(1234), UInt64.valueOf(2345)),
+          keys.get(3),
+              optionalSigningRecord(
+                  UInt64.valueOf(1024000), UInt64.valueOf(12345), UInt64.valueOf(2345)));
+
+  @Test
+  void updateSigningRecord_shouldNotUpdateBetterRecords() {
+    assertThat(
+            SlashingProtectionRepairer.updateSigningRecord(
+                ZERO, ZERO, Optional.of(validatorSigningRecord)))
+        .isEqualTo(validatorSigningRecord);
+  }
+
+  @Test
+  void updateSigningRecord_shouldNotRequireInitialRecord() {
+    assertThat(SlashingProtectionRepairer.updateSigningRecord(ONE, ONE, Optional.empty()))
+        .isEqualTo(validatorSigningRecord);
+  }
+
+  @Test
+  void updateSigningRecord_shouldUpdateBlockSlot() {
+    final ValidatorSigningRecord expectedValue = new ValidatorSigningRecord(null, TWO, ONE, ONE);
+    assertThat(
+            SlashingProtectionRepairer.updateSigningRecord(
+                TWO, ZERO, Optional.of(validatorSigningRecord)))
+        .isEqualTo(expectedValue);
+  }
+
+  @Test
+  void updateSigningRecord_shouldUpdateAttestationEpoch() {
+    final ValidatorSigningRecord expectedValue = new ValidatorSigningRecord(null, ONE, TWO, TWO);
+    assertThat(
+            SlashingProtectionRepairer.updateSigningRecord(
+                ONE, TWO, Optional.of(validatorSigningRecord)))
+        .isEqualTo(expectedValue);
+  }
+
+  @Test
+  void updateSigningRecord_shouldUpdateSourceAttestationEpoch() {
+    final ValidatorSigningRecord initialValue = new ValidatorSigningRecord(null, ONE, ZERO, TWO);
+    final ValidatorSigningRecord expectedValue = new ValidatorSigningRecord(null, ONE, ONE, TWO);
+    assertThat(SlashingProtectionRepairer.updateSigningRecord(ONE, ONE, Optional.of(initialValue)))
+        .isEqualTo(expectedValue);
+  }
+
+  @Test
+  void updateSigningRecord_shouldUpdateTargetAttestationEpoch() {
+    final ValidatorSigningRecord initialValue = new ValidatorSigningRecord(null, ONE, TWO, ONE);
+    final ValidatorSigningRecord expectedValue = new ValidatorSigningRecord(null, ONE, TWO, TWO);
+    assertThat(SlashingProtectionRepairer.updateSigningRecord(ZERO, TWO, Optional.of(initialValue)))
+        .isEqualTo(expectedValue);
+  }
+
+  @Test
+  public void shouldNotUpdateFilesWithInvalidPubkeys(@TempDir Path tempDir) throws IOException {
+    final File slashingFolder =
+        new File(tempDir.toAbsolutePath().toFile(), "validator/slashprotection");
+    final Path slashingProtectionPath = slashingFolder.toPath();
```

Comment:
You don't actually need to create these subdirectories.  Just create everything directly in `tempDir`.